projects/cm_mcu/LocalTasks.c

```diff
@@ -30,6 +31,452 @@
+    .n_commands = NCOMMANDS_FFL12_F2,
+    .n_values = NSUPPLIES_FFL12_F2 * NPAGES_FFL12_F2 * NCOMMANDS_FFL12_F2,
+    .n_pages = NPAGES_FFL12_F2,
+    .sm_values = ffl12_f2_values,
+    .smbus = &g_sMaster3,
+    .smbus_status = &eStatus3,
+    .xSem = NULL,
+    .requirePower = true,
+    .stack_size = 4096U,
+};
+
+// Clock arguments for monitoring task
+
+struct dev_moni2c_addr_t clk_moni2c_addrs[] = {
+    {"r0b", 0x70, 1, 0x6b}, // CLK R0B : Si5395-REVA **no info on EEPROM and need .csv file from Charlie**
+    {"r1a", 0x70, 2, 0x6b}, // CLK R1A : Si5395-REVA
+    {"r1b", 0x70, 3, 0x6b}, // CLK R1B : Si5395-REVA
+    {"r1c", 0x70, 4, 0x6b}, // CLK R1C : Si5395-REVA **no info on EEPROM and need .csv file from Charlie**
+};
+
+struct sm_command_t sm_command_clk[] = {
+    // device information on page 0 : table 16.2 and 16.4
+    {1, 0x00, 0x02, 1, "PN_BASE", 0xff, "", SM_STATUS},    // page 0x00
+    {1, 0x00, 0x03, 1, "PN_BASE", 0xff, "", SM_STATUS},    // page 0x00
+    {1, 0x00, 0x05, 1, "DEVICE_REV", 0xff, "", SM_STATUS}, // page 0x00
+    {1, 0x00, 0x0B, 1, "I2C_ADDR", 0x7f, "", SM_STATUS},   // page 0x00 **for testing**
+    // internal statuses on page 0 : table 16.8 and 16.9
+    {1, 0x00, 0x0C, 1, "LOSXAXB", 0x02, "", SM_STATUS},   // page 0x00
+    {1, 0x00, 0x0D, 1, "LOSOFF_IN", 0xff, "", SM_STATUS}, // page 0x00
+    /*
+    {1, 0x00, 0x0C, 1, "LOSXAXB", 1, 1, "", SM_STATUS}, // page 0x00
+    {1, 0x00, 0x0D, 1, "LOS_IN0", 0, 0, "", SM_STATUS}, // page 0x00
+    {1, 0x00, 0x0D, 1, "LOS_IN1", 1, 1, "", SM_STATUS}, // page 0x00
+    {1, 0x00, 0x0D, 1, "LOS_IN2", 2, 2, "", SM_STATUS}, // page 0x00
+    {1, 0x00, 0x0D, 1, "LOS_IN3", 3, 3, "", SM_STATUS}, // page 0x00
+    {1, 0x00, 0x0D, 1, "OOF_IN0", 4, 4, "", SM_STATUS}, // page 0x00
+    {1, 0x00, 0x0D, 1, "OOF_IN1", 5, 5, "", SM_STATUS}, // page 0x00
+    {1, 0x00, 0x0D, 1, "OOF_IN2", 6, 6, "", SM_STATUS}, // page 0x00
+    {1, 0x00, 0x0D, 1, "OOF_IN3", 7, 7, "", SM_STATUS}, // page 0x00
+    */
+    {1, 0x00, 0x0E, 1, "LOL", 0x02, "", SM_STATUS}, // page 0x00
+};
+
+uint16_t clk_values[NSUPPLIES_CLK * NPAGES_CLK * NCOMMANDS_CLK];
+
+struct MonitorI2CTaskArgs_t clock_args = {
+    .name = "CLKSI",
+    .devices = clk_moni2c_addrs,
+    .i2c_dev = I2C_DEVICE_CLK,
+    .n_devices = NSUPPLIES_CLK,
+    .commands = sm_command_clk,
+    .n_commands = NCOMMANDS_CLK,
+    .n_values = NSUPPLIES_CLK * NPAGES_CLK * NCOMMANDS_CLK,
+    .n_pages = NPAGES_CLK,
+    .sm_values = clk_values,
+    .smbus = &g_sMaster2,
+    .smbus_status = &eStatus2,
+    .xSem = NULL,
+    .requirePower = true,
+    .stack_size = 4096U,
+};
+
+struct dev_moni2c_addr_t clkr0a_moni2c_addrs[] = {
+    {"r0a", 0x70, 0, 0x77}, // CLK R0A : Si5341-REVD
+};
+
+struct sm_command_t sm_command_clkr0a[] = {
+    // device information on page 0 : table 14.4 and 14.6
+    {1, 0x00, 0x02, 1, "PN_BASE", 0xff, "", SM_STATUS},    // page 0x00
```

Comment:
I thought we were going to remove this duplicated read of the same register

Reply:
I thought you meant only those 8 bits with the same register. I think reading the same register 5x of only r0a with decoded information is not a bad idea but I can do whichever way you would suggest.

Reply:
please only read the register 1x

Reply:
Okay, Done.